src/ansys/mapdl/core/launcher.py

```diff
@@ -2023,6 +2067,155 @@ def _parse_ip_route(output):
+        # Reference: https://stackoverflow.com/a/51141287/6650211
+        # I'm assuming the env var makes sense.
+        #
+        # - SLURM_CPUS_ON_NODE is a property of the cluster, not of the job.
+        #
+        options = [
+            # 4,  # Fall back option
+            SLURM_CPUS_PER_TASK * SLURM_NTASKS,  # (CPUs)
+            SLURM_NPROCS,  # (CPUs)
+            # SLURM_NTASKS,  # (tasks) Not necessary the number of CPUs,
+            # SLURM_NNODES * SLURM_TASKS_PER_NODE * SLURM_CPUS_PER_TASK,  # (CPUs)
+            SLURM_CPUS_ON_NODE * SLURM_NNODES,  # (cpus)
+        ]
+        LOG.info(f"On SLURM number of processors options {options}")
+        nproc = max(options)
+
+    LOG.info(f"Setting number of CPUs to: {nproc}")
+
+    if not ram:
+        if SLURM_MEM_PER_NODE:
+            # RAM argument is in MB, so we need to convert
+
+            if SLURM_MEM_PER_NODE[-1] == "T":  # tera
+                ram = int(SLURM_MEM_PER_NODE[:-1]) * (2**10) ** 2
+            elif SLURM_MEM_PER_NODE[-1] == "G":  # giga
+                ram = int(SLURM_MEM_PER_NODE[:-1]) * (2**10) ** 1
+            elif SLURM_MEM_PER_NODE[-1].upper() == "k":  # kilo
+                ram = int(SLURM_MEM_PER_NODE[:-1]) * (2**10) ** (-1)
+            else:  # Mega
+                ram = int(SLURM_MEM_PER_NODE)
+
+    LOG.info(f"Setting RAM to: {ram}")
+
+    # We use "-dis " (with space) to avoid collision with user variables such
+    # as `-distro` or so
+    if "-dis " not in additional_switches and not additional_switches.endswith("-dis"):
+        additional_switches += " -dis"
+
+    ## Getting the node list
+    machines = ""
+    # parsing nodes to list
+    if SLURM_NODELIST:
+        try:
+            p = subprocess.Popen(
+                ["scontrol", "show", "hostnames", f"{SLURM_NODELIST}"],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+            )
+            stderr = p.stderr.read().decode()
+            stdout = p.stdout.read().decode()
+
+            if "Invalid hostlist" in stderr:
+                raise ValueError(
+                    "The node list is invalid, or it could not be parsed.\n",
+                    "Are you passing the nodes correctly?\n",
+                    f"Nodes list: {SLURM_NODELIST}",
+                )
+            if stderr:
+                raise RuntimeError(stderr)
+            nodes = stdout.strip().splitlines()
+
+            machines = ":".join([f"{each_node}" for each_node in nodes])
+
+            # The following code creates the cmd line bit for MAPDL. It seems it
+            # is not needed in slurm.
+            # machines = " -machines " + ":".join([
+            #     f"{each_node}:{SLURM_CPUS_ON_NODE}" for each_node in nodes
+            # ])
+
+            # We do not need to inject the machines in MAPDL command line.
+            # additional_switches += machines
+            LOG.info(f"Using nodes configuration: {machines}")
+
```

Comment:
I don't see where the launcher actually submits the job to slurm

Reply:
That's something that the person needs to do:

```console
sbatch my_python_script.py
```
where ``my_python_script.py`` is:
```py
from ansys.mapdl.core import launch_mapdl

mapdl = launch_mapdl() # This will read the job env vars

mapdl.prep7()
# ...
# 
mapdl.exit()
```

This PR is focusing on "full batch" mode... where you have nothing running on the VDI or "client". 

The python client is also running on the computation nodes.

For having python running on the client, and MAPDL in the computation nodes, another PR will be needed. But hopefully based on PyHPS.

Reply:
@germa89 got it, this will require some documentation perhaps. but it can wait

Reply:
I agree... x2